parser-typechecker/src/Unison/CommandLine/Main.hs

```diff
@@ -184,13 +185,15 @@ main dir initialPath initialInputs startRuntime codebase = do
       awaitInput = do
         -- use up buffered input before consulting external events
         i <- readIORef initialInputsRef
-        case i of
+        (case i of
           h:t -> writeIORef initialInputsRef t >> pure h
           [] ->
             -- Race the user input and file watch.
             Async.race (atomically $ Q.peek eventQueue) getInput >>= \case
               Left _ -> Left <$> atomically (Q.dequeue eventQueue)
-              x      -> pure x
+              x      -> pure x) `catch` interruptHandler
+      interruptHandler (asyncExceptionFromException -> Just UserInterrupt) = awaitInput
+      interruptHandler _ = pure $ Right QuitI
```

Comment:
I think it would probably be more correct to re-throw non `UserInterrupt` exceptions rather than just quit.

Reply:
Should we wait on this?

Reply:
I think it could be fixed up post-merge, it's good enuf ;)